scipy/stats/_multivariate.py

```diff
@@ -5148,8 +5150,8 @@ def __call__(self, dim=None, seed=None):
     def _process_parameters(self, dim):
         """Dimension N must be specified; it cannot be inferred."""
         if dim is None or not np.isscalar(dim) or dim < 1 or dim != int(dim):
```

Comment:
If you don't want to use `np.issubdtype(dim, np.integer)`, `int(dim)` needs to be returned. Currently this fails:
```python3
from scipy.stats import random_direction
random_direction(3.0).rvs()
```
with
```
TypeError: 'numpy.float64' object cannot be interpreted as an integer
```